src/app/pages/ui/ag-grid/ag-grid.service.ts

```diff
@@ -0,0 +1,240 @@
+    {
+      id: '7',
+      title: 'Rebecca',
+      author: 'Daphne du Maurier',
+      genre: [
+        'Gothic fiction',
+      ],
+      year: '2016',
+      published: '1938',
+      language: 'English',
+      originalLanguage: 'English',
+      count: 2,
+    },
+    {
+      id: '8',
+      title: 'Dracula',
+      author: 'Bram Stoker',
+      genre: [
+        'Gothic fiction',
+        'Horror fiction',
+      ],
+      year: '2015',
+      published: '1897',
+      language: 'English',
+      originalLanguage: 'English',
+      count: 0,
+    },
+    {
+      id: '9',
+      title: 'The Little Prince',
+      author: 'Antoine de Saint-Exupery',
+
+      genre: ['Fiction'],
+      year: '2015',
+      published: '1943',
+      language: 'English',
+      originalLanguage: 'French',
+      count: 1,
+    },
+    {
+      id: '10',
+      title: 'Les Miserables',
+      author: 'Victor Hugo',
+      genre: [
+        'Novel',
+        'Fiction',
+      ],
+      year: '2014',
+      published: '1862',
+      language: 'English',
+      originalLanguage: 'French',
+      count: 3,
+    },
+    {
+      id: '11',
+      title: 'Alice in Wonderland',
+      author: 'Lewis Carroll',
+      genre: [
+        'Children\'s book',
+        'Fantasy',
+      ],
+      year: '2018',
+      published: '1865',
+      language: 'English',
+      originalLanguage: 'English',
+      count: 4,
+    },
+    {
+      id: '12',
+      title: 'Anne of Green Gables',
+      author: 'L.M. Montgomery',
+      genre: ['Novel',
+      ],
+      year: '2019',
+      published: '1908',
+      language: 'English',
+      originalLanguage: 'English',
+      count: 3,
+    },
+    {
+      id: '13',
+      title: 'The Hitchhiker\'s Guide to the Galaxy',
+      author: 'Douglas Adams',
+      genre: [
+        'Adventure',
+        'Fiction',
+      ],
+      year: '2017',
+      published: '1979',
+      language: 'English',
+      originalLanguage: 'English',
+      count: 3,
+    }, {
+      id: '14',
+      title: 'Madame Bovary',
+      author: 'Gustave Flaubert',
+      genre: ['Novel'],
+      year: '2018',
+      published: '1856',
+      language: 'English',
+      originalLanguage: 'French',
+      count: 2,
+    },
+  ];
+
+  private genreColors = {
+    'Children\'s book': 'orange',
+    'Gothic fiction': 'purple',
+    'Horror fiction': 'dark-gray',
+    'Fiction': 'green',
+    'Novel': 'teal',
+    'Fantasy': 'red',
+    'Adventure': 'light-blue',
+  };
+
+  private elementTypeColors = {
+    'non metal': 'smooth-gray',
+    'noble gas': 'cerulean',
+    'alkali metal': 'amber',
+    'alkaline earth metal': 'red',
+    'halogen': 'baby-blue',
+    'metalloid': 'mint',
+    'metal': 'green',
+  };
+
+  constructor(private http: HttpClient) {
+  }
+
+  getGropedData(count: number = 40): Observable<any[]> {
+    return this.http.get<any[]>('https://raw.githubusercontent.com/ag-grid/ag-grid/master/packages/ag-grid-docs/src/olympicWinnersSmall.json')
```

Comment:
It will be better to move this file to the repository (and maybe make it smaller)